contracts/Treasury.sol

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import '@openzeppelin/contracts/utils/Address.sol';
+import '@openzeppelin/contracts/utils/Context.sol';
+import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+
+interface IERC20Allowance is IERC20 {
+    function increaseAllowance(address spender, uint256 addedValue)
+        external
+        returns (bool);
+
+    function decreaseAllowance(address spender, uint256 subtractedValue)
+        external
+        returns (bool);
+}
+
+// Treasury contract.
+contract Treasury is Context {
+    address public governance;
+    address public executor;
+
+    mapping(address => uint256) public balances;
+    bytes32 public name = 'TREASURY';
+
+    event Received(address from, address asset, uint256 amount);
+    event Sent(address to, address asset, uint256 amount);
+    event IncreasedAllowance(address spender, address asset, uint256 amount);
+    event DecreasedAllowance(address spender, address asset, uint256 amount);
+
+    modifier onlyGovernance() {
```

Comment:
This modifier is not being used, is it needed? ...could the `governance` field also be removed?